### PR TITLE
[dbnode] Add ability to set aggregate limits

### PR DESCRIPTION
**Pull request by arnikola**
Adds ability to set global limits on aggregate docs count

### CODE REVIEW

File: src/dbnode/storage/limits/query_limits.go [hunk 1]
```diff
@@ -106,6 +108,10 @@ func NewQueryLimits(options Options) (QueryLimits, error) {
 			sourceLoggerBuilder, map[string]string{"type": "aggregate"})
 	)
 
+	// NB: All metrics for aggregate methods should show up with the same metric
```
Comment (ryanhall07) [inline src/dbnode/storage/limits/query_limits.go:111]: why don't we just stop updating `docsMatched` for the aggregate path so it essentially becomes docs matched for the fetch path.

also, don't follow why you need to update the name after constructing.

File: src/dbnode/storage/limits/types.go [hunk 2]
```diff
@@ -39,13 +39,13 @@ type QueryLimits interface {
 	DocsLimit() LookbackLimit
```
Comment (ryanhall07) [inline src/dbnode/storage/limits/types.go:39]: can you update the `DocsLimit` to make it clear it's for fetch?

Comment (ryanhall07) [reply src/dbnode/storage/limits/types.go:39]: also can we just rename the code refs to be `FetchDocsLimit`?

File: src/cmd/services/m3dbnode/config/limits.go [hunk 3]
```diff
@@ -40,6 +40,11 @@ type LimitsConfiguration struct {
 	// max is surpassed encounter an error.
 	MaxRecentlyQueriedSeriesBlocks *MaxRecentQueryResourceLimitConfiguration `yaml:"maxRecentlyQueriedSeriesBlocks"`
 
+	// MaxRecentlyQueriedMetadata sets the upper limit on metatadata counts
```
Comment (wesleyk) [inline src/cmd/services/m3dbnode/config/limits.go:43]: metadata

File: src/dbnode/storage/limits/types.go [hunk 4]
```diff
@@ -39,13 +39,13 @@ type QueryLimits interface {
...
 	// BytesReadLimit limits queries by a global concurrent count of bytes read from disk.
 	BytesReadLimit() LookbackLimit
+	// AnyFetchExceeded returns an error if any of the query limits are exceeded.
```
Comment (wesleyk) [inline src/dbnode/storage/limits/types.go:42]: can we update the comment here to specify it's only for fetch-related limits?

File: src/dbnode/storage/index/block_test.go [hunk 5]
```diff
@@ -1990,6 +1991,88 @@ func TestBlockAggregate(t *testing.T) {
...
+	}
+	iter.EXPECT().Close(gomock.Any()).Return(nil)
+	iter.EXPECT().SearchDuration().Return(time.Second)
+
+	exhaustive, err := b.Aggregate(
+		ctx,
+		QueryOptions{SeriesLimit: seriesLimit},
+		results,
+		emptyLogFields)
+	require.Error(t, err)
```
Comment (wesleyk) [inline src/dbnode/storage/index/block_test.go:2060]: can we assert on the error being limit-related?

File: src/dbnode/storage/limits/query_limits.go (hunk 1 above)
Comment (arnikola) [reply src/dbnode/storage/limits/query_limits.go:111]: We want them to appear in the same metric, but under different tags, so the generated metrics for both aggregate and fetch limits create metrics here. The name is updated to affect the error message received when the limit is tripped

**Review by wesleyk — APPROVED** [review]
LG with comments

**Pull request merged**